Compatibility Stable 25-1 -> Main(or stable-25-1-analyctis) : Verify with Date32: TActiveTransactionZone(): requirement !Self->ActiveTransaction failed

версия: 25-1  с последующим обновление до main
Получил в тесте 
```./ya make -ttt --build "relwithdebinfo" -F 'test_compatibility.py::TestCompatibility::test_tpch1[last_stable_to_current-column]' ydb/tests/functional/compatibility```

ЕСЛИ ЗАПУСКАЕШЬ ЭТОТ ТЕСТ ТО ЗАКОММЕНТИРУЙ ЭТУ СТРОЧКУ и будет с Date32
https://github.com/naspirato/ydb/blob/400daebd1dffbb3f27574a92f88b65b09e9172ac/ydb/tests/functional/compatibility/test_compatibility.py#L282

1) создать таблицу с Date32 (Можно попробовать через cli tpch) + налить данных (через tpch import)
2) обновить кластер до main 
3) выполнить запрос напирмер `Select version()`

вызов cli для создания таблицы
```
init_command = [
            yatest.common.binary_path(os.getenv("YDB_CLI_BINARY")),
            "--verbose",
            "--endpoint",
            "grpc://localhost:%d" % self.cluster.nodes[1].port,
            "--database=/Root",
            "workload",
            "tpch",
            "-p",
            "tpch",
            "init",
            "--store=column",
            "--partition-size=25",
        ]
```
для импорта
```
      import_command = [
            yatest.common.binary_path(os.getenv("YDB_CLI_BINARY")),
            "--verbose",
            "--endpoint",
            "grpc://localhost:%d" % self.cluster.nodes[1].port,
            "--database=/Root",
            "workload",
            "tpch",
            "-p",
            "tpch",
            "import",
            "generator",
            "--scale=1",
```

```
ydb/tests/library/harness/kikimr_runner.py:499: in stop
    raise daemon.SeveralDaemonErrors(saved_exceptions)
E   ydb.tests.library.harness.daemon.SeveralDaemonErrors: Daemon failed with message: Unexpectedly finished before stop.
E   Process exit_code = -6.
E   Stdout file name: 
E   /home/kirrysin/.ya/build/build_root/zq46/00000a/ydb/tests/functional/compatibility/test-results/py3test/testing_out_stuff/test_compatibility.py.TestCompatibility.test_tpch1.last_stable_to_current-column/cluster/node_7/stdout
E   Stderr file name: 
E   /home/kirrysin/.ya/build/build_root/zq46/00000a/ydb/tests/functional/compatibility/test-results/py3test/testing_out_stuff/test_compatibility.py.TestCompatibility.test_tpch1.last_stable_to_current-column/cluster/node_7/stderr
E   Stderr content:
E   
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   Current KQP shutdown state: spent 1e-06 seconds, not started yet
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   VERIFY failed (2025-04-22T18:49:10.723281+0300):
E   ydb/core/tablet_flat/flat_executor.cpp:104
E   TActiveTransactionZone(): requirement !Self->ActiveTransaction failed
E   0. /-S/util/system/yassert.cpp:83: NPrivate::InternalPanicImpl(int, char const*, char const*, int, int, int, TBasicStringBuf<char, std::__y1::char_traits<char>>, char const*, unsigned long) @ 0xA1F7E95
E   1. /-S/util/system/yassert.cpp:55: NPrivate::Panic(NPrivate::TStaticBuf const&, int, char const*, char const*, char const*, ...) @ 0xA1F0BFC
E   2. /-S/ydb/core/tablet_flat/flat_executor.cpp:104: NKikimr::NTabletFlatExecutor::TExecutor::TActiveTransactionZone::TActiveTransactionZone(NKikimr::NTabletFlatExecutor::TExecutor*) @ 0xCB937B9
E   3. /-S/ydb/core/tablet_flat/flat_executor.cpp:1894: NKikimr::NTabletFlatExecutor::TExecutor::ExecuteTransaction(NKikimr::NTabletFlatExecutor::TSeat*) @ 0xCB937B9
E   4. /-S/ydb/core/tablet_flat/flat_executor.cpp:4143: NKikimr::NTabletFlatExecutor::TExecutor::StateWork(TAutoPtr<NActors::IEventHandle, TDelete>&) @ 0xCB7E616
E   5. /-S/ydb/library/actors/core/actor.cpp:280: NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) @ 0xAF23EB4
E   6. /-S/ydb/library/actors/core/executor_thread.cpp:269: NActors::TExecutorThread::Execute(NActors::TMailbox*, bool) @ 0xAF5BAD0
E   7. /-S/ydb/library/actors/core/executor_thread.cpp:460: NActors::TExecutorThread::ProcessExecutorPool()::$_0::operator()(NActors::TMailbox*, bool) const @ 0xAF5F5D1
E   8. /-S/ydb/library/actors/core/executor_thread.cpp:512: NActors::TExecutorThread::ProcessExecutorPool() @ 0xAF5F220
E   9. /-S/ydb/library/actors/core/executor_thread.cpp:538: NActors::TExecutorThread::ThreadProc() @ 0xAF5FD9A
E   10. /-S/util/system/thread.cpp:244: (anonymous namespace)::TPosixThread::ThreadProxy(void*) @ 0xA1F8CEC
E   11. /build/glibc-FcRMwW/glibc-2.31/nptl/pthread_create.c:477: start_thread @ 0x7FEFA8F46608
E   12. ../sysdeps/unix/sysv/linux/x86_64/clone.S:95: ?? @ 0x7FEFA8E6B352
E   Daemon failed with message: Unexpectedly finished before stop.
E   Process exit_code = -6.
E   Stdout file name: 
E   /home/kirrysin/.ya/build/build_root/zq46/00000a/ydb/tests/functional/compatibility/test-results/py3test/testing_out_stuff/test_compatibility.py.TestCompatibility.test_tpch1.last_stable_to_current-column/cluster/node_5/stdout
E   Stderr file name: 
E   /home/kirrysin/.ya/build/build_root/zq46/00000a/ydb/tests/functional/compatibility/test-results/py3test/testing_out_stuff/test_compatibility.py.TestCompatibility.test_tpch1.last_stable_to_current-column/cluster/node_5/stderr
E   Stderr content:
E   
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   Current KQP shutdown state: spent 0 seconds, not started yet
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   VERIFY failed (2025-04-22T18:49:18.374798+0300):
E   ydb/core/tablet_flat/flat_executor.cpp:104
E   TActiveTransactionZone(): requirement !Self->ActiveTransaction failed
E   0. /-S/util/system/yassert.cpp:83: NPrivate::InternalPanicImpl(int, char const*, char const*, int, int, int, TBasicStringBuf<char, std::__y1::char_traits<char>>, char const*, unsigned long) @ 0xA1F7E95
E   1. /-S/util/system/yassert.cpp:55: NPrivate::Panic(NPrivate::TStaticBuf const&, int, char const*, char const*, char const*, ...) @ 0xA1F0BFC
E   2. /-S/ydb/core/tablet_flat/flat_executor.cpp:104: NKikimr::NTabletFlatExecutor::TExecutor::TActiveTransactionZone::TActiveTransactionZone(NKikimr::NTabletFlatExecutor::TExecutor*) @ 0xCB937B9
E   3. /-S/ydb/core/tablet_flat/flat_executor.cpp:1894: NKikimr::NTabletFlatExecutor::TExecutor::ExecuteTransaction(NKikimr::NTabletFlatExecutor::TSeat*) @ 0xCB937B9
E   4. /-S/ydb/core/tablet_flat/flat_executor.cpp:4143: NKikimr::NTabletFlatExecutor::TExecutor::StateWork(TAutoPtr<NActors::IEventHandle, TDelete>&) @ 0xCB7E616
E   5. /-S/ydb/library/actors/core/actor.cpp:280: NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) @ 0xAF23EB4
E   6. /-S/ydb/library/actors/core/executor_thread.cpp:269: NActors::TExecutorThread::Execute(NActors::TMailbox*, bool) @ 0xAF5BAD0
E   7. /-S/ydb/library/actors/core/executor_thread.cpp:460: NActors::TExecutorThread::ProcessExecutorPool()::$_0::operator()(NActors::TMailbox*, bool) const @ 0xAF5F5D1
E   8. /-S/ydb/library/actors/core/executor_thread.cpp:512: NActors::TExecutorThread::ProcessExecutorPool() @ 0xAF5F220
E   9. /-S/ydb/library/actors/core/executor_thread.cpp:538: NActors::TExecutorThread::ThreadProc() @ 0xAF5FD9A
E   10. /-S/util/system/thread.cpp:244: (anonymous namespace)::TPosixThread::ThreadProxy(void*) @ 0xA1F8CEC
E   11. /build/glibc-FcRMwW/glibc-2.31/nptl/pthread_create.c:477: start_thread @ 0x7F0D9B37E608
E   12. ../sysdeps/unix/sysv/linux/x86_64/clone.S:95: ?? @ 0x7F0D9B2A3352
```

--- a/ydb/core/tx/columnshard/columnshard.cpp
+++ b/ydb/core/tx/columnshard/columnshard.cpp
@@ -119,6 +119,7 @@ void TColumnShard::OnActivateExecutor(const TActorContext& ctx) {
 
     AFL_INFO(NKikimrServices::TX_COLUMNSHARD)("event", "initialize_shard")("step", "initialize_tiring_finished");
     auto& icb = *AppData(ctx)->Icb;
+    SpaceWatcherId = RegisterWithSameMailbox(SpaceWatcher);
     Limits.RegisterControls(icb);
     Settings.RegisterControls(icb);
     ResourceSubscribeActor = ctx.Register(new NOlap::NResourceBroker::NSubscribe::TActor(TabletID(), SelfId()));

--- a/ydb/core/tx/columnshard/columnshard_impl.cpp
+++ b/ydb/core/tx/columnshard/columnshard_impl.cpp
@@ -98,7 +98,6 @@ TColumnShard::TColumnShard(TTabletStorageInfo* info, const TActorId& tablet)
     , SysLocks(this) {
     AFL_VERIFY(TabletActivityImpl->Inc() == 1);
     SpaceWatcher = new TSpaceWatcher(this);
-    SpaceWatcherId = TActorContext::AsActorContext().Register(SpaceWatcher);
 }
 
 void TColumnShard::OnDetach(const TActorContext& ctx) {
@@ -1443,7 +1442,7 @@ public:
 
     bool Execute(TTransactionContext& txc, const TActorContext& /*ctx*/) override {
         NIceDb::TNiceDb db(txc.DB);
-        
+
         TBlobGroupSelector selector(Self->Info());
         bool reask = false;
         NActors::TLogContextGuard lGuard = NActors::TLogContextBuilder::Build()("consumer", Consumer)("event", "TTxAskPortionChunks::Execute");

--- a/ydb/tests/compatibility/test_compatibility.py
+++ b/ydb/tests/compatibility/test_compatibility.py
@@ -88,8 +88,12 @@ class TestCompatibility(RestartToAnotherVersionFixture):
         upsert_and_check_sum(self, iteration_count=2, start_index=100)
         assert self.execute_scan_query('select count(*) as row_count from `sample_table`')[0]['row_count'] == 500, 'Expected 500 rows: update 100-200 rows and added 300 rows'
 
-    @pytest.mark.parametrize("store_type", ["row", "column"])
-    def test_tpch1(self, store_type):
+    @pytest.mark.parametrize("store_type, date_args", [
+        pytest.param("row",    ["--datetime"], id="row"),
+        pytest.param("column", ["--datetime"], id="column"),
+        pytest.param("column", []            , id="column-date64")
+    ])
+    def test_tpch1(self, store_type, date_args):
         result_json_path = os.path.join(yatest.common.test_output_path(), "result.json")
         query_output_path = os.path.join(yatest.common.test_output_path(), "query_output.json")
         init_command = [
@@ -104,9 +108,9 @@ class TestCompatibility(RestartToAnotherVersionFixture):
             "tpch",
             "init",
             "--store={}".format(store_type),
-            "--datetime",  # use 32 bit dates instead of 64 (not supported in 24-4)
             "--partition-size=25",
-        ]
+        ] + date_args  # use 32 bit dates instead of 64 (not supported in 24-4)]
+
         import_command = [
             yatest.common.binary_path(os.getenv("YDB_CLI_BINARY")),
             "--verbose",

--- a/ydb/tests/compatibility/test_stress.py
+++ b/ydb/tests/compatibility/test_stress.py
@@ -180,8 +180,12 @@ class TestStress(MixedClusterFixture):
         yatest.common.execute(init_command, wait=True, stdout=self.output_f, stderr=self.output_f)
         yatest.common.execute(run_command, wait=True, stdout=self.output_f, stderr=self.output_f)
 
-    @pytest.mark.parametrize("store_type", ["row", "column"])
-    def test_tpch1(self, store_type):
+    @pytest.mark.parametrize("store_type, date_args", [
+        pytest.param("row",    ["--datetime"], id="row"),
+        pytest.param("column", ["--datetime"], id="column"),
+        pytest.param("column", []            , id="column-date64")
+    ])
+    def test_tpch1(self, store_type, date_args):
         init_command = [
             yatest.common.binary_path(os.getenv("YDB_CLI_BINARY")),
             "--verbose",
@@ -194,9 +198,8 @@ class TestStress(MixedClusterFixture):
             "tpch",
             "init",
             "--store={}".format(store_type),
-            "--datetime",  # use 32 bit dates instead of 64 (not supported in 24-4)
             "--partition-size=25",
-        ]
+        ] + date_args  # use 32 bit dates instead of 64 (not supported in 24-4)]
         import_command = [
             yatest.common.binary_path(os.getenv("YDB_CLI_BINARY")),
             "--verbose",
@@ -235,8 +238,12 @@ class TestStress(MixedClusterFixture):
         yatest.common.execute(run_command, wait=True, stdout=self.output_f, stderr=self.output_f)
 
     @pytest.mark.skip(reason="Not stabilized yet")
-    @pytest.mark.parametrize("store_type", ["row", "column"])
-    def test_tpcds1(self, store_type):
+    @pytest.mark.parametrize("store_type, date_args", [
+        pytest.param("row",    ["--datetime"], id="row"),
+        pytest.param("column", ["--datetime"], id="column"),
+        pytest.param("column", []            , id="column-date64")
+    ])
+    def test_tpcds1(self, store_type, date_args):
         init_command = [
             yatest.common.binary_path(os.getenv("YDB_CLI_BINARY")),
             "--verbose",
@@ -247,12 +254,10 @@ class TestStress(MixedClusterFixture):
             "tpcds",
             "-p",
             "tpcds",
-
             "init",
             "--store={}".format(store_type),
-            "--datetime",  # use 32 bit dates instead of 64 (not supported in 24-4)
             "--partition-size=25",
-        ]
+        ] + date_args  # use 32 bit dates instead of 64 (not supported in 24-4)]
         import_command = [
             yatest.common.binary_path(os.getenv("YDB_CLI_BINARY")),
             "--verbose",

